File: Programmers/Algorithm_Kit/05_brute_force/004.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

vector<int> solution(int brown, int yellow) {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL); cout.tie(NULL);
    
    vector<int> answer;
    
    int sum = brown + yellow;

    for (int i = 1; i <= sum; i++) {
        if (sum % i == 0) {
            int b = sum / i;
            int y = i;

            // 조건에 맞는 경우 찾기
            if ((b - 2) * (y - 2) == yellow) {
                answer.push_back(b);
                answer.push_back(y);
                break;
            }
        }
    }

    return answer;
}

File: programmers/algorithmkit/05_brute_force/004.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];

        int temp = brown + yellow;

        for (int i = temp / 3; i >= 3; i--) {
            int width = i;
            int height = temp / i;

            int y = (width - 2) * (height - 2);
            int b = temp - y;

            if (b == brown && y == yellow) {
                answer[0] = width;
                answer[1] = height;

                break;
            }
        }

        return answer;
    }
}

// 탈출 로직이 필수 요소인 문제. break를 사용하지 않으면 가로, 세로가 변경될 수 있다